packages/reason-css-parser/test/Spec.re

```diff
@@ -0,0 +1,99 @@
+/* and _webkit_mask_clip_style = [%value.rec "'border' | 'border-box' | 'padding' | 'padding-box' | 'content' | 'content-box' | 'text'"] */
+
+type webkit_mask_clip_style = [
+  | `border
+  | `border_box
+  | `padding
+  | `padding_box
+  | `content
+  | `content_box
+];
+
+type length;
+type percentage;
+
+/* and property_width = [%value.rec "'auto' | <length> | <percentage> | 'min-content' | 'max-content' | 'fit-content' | fit-content( <length> | <percentage> )"] */
+
+type property_width = [
+  | `auto
+  | `length(length)
+  | `percentage(percentage)
+  | `min_content
+  | `max_content
+  | `fit_content
+  | `fit_content_length(length)
+  | `fit_content_percentage(percentage)
+];
+
+/* and final_bg_layer = [%value.rec "<'background-color'> || <bg-image> || <bg-position> [ '/' <bg-size> ]? || <repeat-style> || <attachment> || <box> || <box>"] */
+
+type xor('a, 'b) = [ | `A('a) | `B('b) ];
+type xor3('a, 'b, 'c) = [ | `A('a) | `B('b) | `C('c)];
+type xor7('a, 'b, 'c, 'd, 'e, 'f, 'g) = [ | `A('a) | `B('b) | `C('c) | `D('d) | `E('e) | `F('f) | `G('g) ];
```

Comment:
Hey, shouldn't this be `or`? 
https://developer.mozilla.org/pt-BR/docs/Web/CSS/Value_definition_syntax#double_bar

Reply:
As the link above describe all the entities are optionals but at least one must be present, how would we encode that? `option` + raise if everything is `None`?

Reply:
Totally, that's an or.

We can't ensure it via the types right?